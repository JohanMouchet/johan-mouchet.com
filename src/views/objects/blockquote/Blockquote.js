// @flow

import * as React from "react";
import cx from "classnames";
import "./___.scss";

/**
 * Blockquote
 *
 * @param {string} quote                        - Blockquote quote
 * @param {string} [outerQuote]                 - Blockquote outer quote
 * @param {string} [cite]                       - Blockquote cite
 * @param {string} [metadata.contextClass]      - Blockquote context class
 * @param {array}  [metadata.modifierClasses]   - Blockquote modifiers classes
 * @param {array}  [metadata.additionalClasses] - Blockquote additional classes
 */

const ___ = (props: Props) => {
  const {
    prop1,
    prop2
  } = props;

  const ___Class = cx(
    "___",
    {
      prop1: "___--prop1"
    }
  );

  return (
	{% set contextClass = props.metadata.contextClass %}
	{% set modifierClasses = props.metadata.modifierClasses %}
	{% set additionalClasses = props.metadata.additionalClasses %}

	<blockquote className="{ contextClass } o-blockquote { 'o-blockquote'|modifierClasses(modifierClasses) } wow { additionalClasses|additionalClasses }">
		<div className="{ 'o-blockquote__quotes'|contextClass(contextClass) }">
			{ props.quote|raw }
		</div>

		{ props.outerQuote|raw }

		{% if props.cite %}
			<cite className="{ 'o-blockquote__cite'|contextClass(contextClass) }">
					{ props.cite|raw }
			</cite>
		{% endif %}
	</blockquote>
  );
};

___.defaultProps = {
  prop1: 2
};

export default ___;
