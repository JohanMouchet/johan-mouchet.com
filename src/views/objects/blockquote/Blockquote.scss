@import "assets/styles/settings/_settings";
@import "assets/styles/tools/_tools";

.o-blockquote {
  margin: t-vertical-rhythm(2, "em", t-modular-scale(1));
  padding-left: 1em;
  font-size: t-modular-scale(1);
  color: $s-color-secondary;
  position: relative;

  &::before {
    display: inline-block;
    content: "";
    height: 100%;
    width: 2px;
    background: $s-color-gray;
    position: absolute;
    left: 0;
    transform-origin: 50% 0;
    animation: blockquote-animation 0.5s ease-out both;

    @include t-wekkit-hdpi-small-dimension("width");

    @keyframes blockquote-animation {
      from {
        transform: scaleY(0);
      }
    }
  }

  > * {
    animation: 0.25s 0.25s ease-out both;
    animation-name: u-animation--fade-in;
  }

  &__quotes {
    margin: 0;
    text-align: inherit;
    quotes: "\201c""\201d";

    &::before,
    &::after {
      font-family: $s-font-family-serif;
      font-weight: 700;
      font-size: t-modular-scale(2);
      line-height: 1;
    }

    &::before {
      content: open-quote;
      margin-right: 0.2em;
    }

    &::after {
      content: close-quote;
      margin-left: 0.2em;
    }

    [href]:hover {
      text-decoration: none;
    }
  }

  /** Pulled **/
  &--pulled {
    float: left;

    @media (min-width: map-get($s-breakpoints, md)) {
      width: 30%;
    }
  }

  /** Cite **/
  &__cite {
    display: block;
    margin-top: t-vertical-rhythm(0.5);
    font-size: t-modular-scale(-2);
    color: $s-color-text-dimmed;

    /* stylelint-disable-next-line no-descending-specificity */
    [href] {
      @include t-link($s-color-text-dimmed);
    }

    &::before {
      color: inherit;
      content: "\2014";
      margin-right: 0.5rem;
    }
  }
}
