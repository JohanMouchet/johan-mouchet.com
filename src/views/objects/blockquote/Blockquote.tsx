import React from "react";
import cx from "classnames";
import styles from "./Blockquote.module.scss";

type Props = {
  quote: React.ReactNode;
  cite?: string;
  pulled?: boolean;
  className?: string | string[] | { [key: string]: boolean };
};

const Blockquote: React.FC<Props> = ({ quote, cite, pulled, className }) => {
  const classNames = cx(
    "o-blockquote",
    className,
    pulled && "o-blockquote--pulled"
  );

  return (
    <blockquote className={classNames}>
      <div className="o-blockquote__quotes u-animation u-animation--fade-in">
        {quote}
      </div>
      {cite && (
        <cite className="o-blockquote__cite u-animation u-animation--fade-in">
          {cite}
        </cite>
      )}
    </blockquote>
  );
};

export default Blockquote;
