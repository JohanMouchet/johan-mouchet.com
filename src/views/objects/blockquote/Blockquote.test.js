import React from "react";
import { shallow } from "enzyme";
import Blockquote from ".";

it("Renders <Blockquote /> without crashing", () => {
  shallow(<Blockquote />);
});
