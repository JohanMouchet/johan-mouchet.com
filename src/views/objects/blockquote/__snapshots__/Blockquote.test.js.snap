// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders <Blockquote /> without crashing 1`] = `
<blockquote
  className="o-blockquote"
>
  <div
    className="o-blockquote__quotes"
  />
</blockquote>
`;
